src/main/java/backend/model/sponsoring/Sponsoring.kt

```diff
@@ -97,10 +104,15 @@ class Sponsoring : BasicEntity, Billable {
         this.registeredSponsor = null
     }
 
+    @Suppress("UNUSED") //Used by Spring @PreAuthorize
+    fun isTeamMember(username: String): Boolean {
+        return this.teams?.any { it.isMember(username) }
+    }
+
     @Suppress("UNUSED") //Used by Spring @PreAuthorize
     fun checkWithdrawPermissions(username: String): Boolean {
         return when {
-            this.unregisteredSponsor != null -> this.team!!.isMember(username)
+            this.unregisteredSponsor != null -> this.unregisteredSponsor!!.team?.isMember(username) == true
```

Comment:
I'm a bit unsure whether this is correct. Can the one who created the sponsoring (so `unregistedSponsor.team` do this or any of the teams in the `teams` collection? Ideally it's the same team anyway.

We need to check this.